### ISSUE TITLE
nansum op 在 graph global 下处理 0 size tensor 报错

### ISSUE THREAD

**Issue by lixiang007666** [Oneflow-Inc/oneflow#10067]
在支持 graph global test 项目的过程中，遇到下面的 case 两卡测试 Segmentation fault。

```python
@autotest(n=1, check_graph=True)
def _test_global_nansum_with_0_size_tensor(test_case, placement, sbp):
    x = random_tensor(4, 8, 16, 0, 24).to_global(placement, sbp)
    mask = torch.ones_like(x).bool()
    x = x.masked_fill(mask, float("nan"))
    y = torch.nansum(x, dim=random(0, 3).to(int))
    return y
```

最小复现代码：

```python
import os
import sys

# os.environ["MASTER_ADDR"] = "127.0.0.1"
# os.environ["MASTER_PORT"] = "29007"
# os.environ["WORLD_SIZE"] = "2"
# os.environ["RANK"] = sys.argv[1]
# os.environ["LOCAL_RANK"] = sys.argv[1]
os.environ["ONEFLOW_DEBUG_MODE"] = "1"

import oneflow as flow
import oneflow.nn as nn

placement = flow.placement("cuda", ranks=[0, 1])
sbp = flow.sbp.split(dim = 0) # sbp 划分的维度
sbp2 = flow.sbp.broadcast

x = flow.randn(4, 8, 0, 16, 24).to_global(placement = placement, sbp = sbp)

class Module(nn.Module):
    def __init__(self):
        super().__init__()

    def forward(self, x):
        mask = flow.ones_like(x).bool()
        x = x.masked_fill(mask, float("nan"))
        print(x)
        y = flow.nansum(x, dim = 0) # nansum reduce 的维度
        return y

model = Module().to_global(placement = placement, sbp = sbp2)

class Graph(nn.Graph):
  def __init__(self):
    super().__init__()
    self.model = model

  def build(self, input):
    return self.model(input)

graph = Graph()
print(graph(x))

```

复现这个错误的条件是：

- nansum reduce 的维度和 sbp 划分的维度相同（上面的例子中都是 dim 0）。
- 0 size tensor。
- graph global。

<details>
  <summary>gdb output</summary>
  <pre><code> 
Thread 1 "python3" received signal SIGSEGV, Segmentation fault.
0x00007fffd8743782 in oneflow::SliceBoxingOp::InferOutBlobDescs(std::function<oneflow::BlobDesc* (std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)> const&, oneflow::ParallelContext const*) const ()
   from /data/home/lixiang/oneflow/build/liboneflow.so
(gdb) bt
#0  0x00007fffd8743782 in oneflow::SliceBoxingOp::InferOutBlobDescs(std::function<oneflow::BlobDesc* (std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)> const&, oneflow::ParallelContext const*) const ()
   from /data/home/lixiang/oneflow/build/liboneflow.so
#1  0x00007fffd870840f in oneflow::Operator::InferOutBlobDescsIf(std::function<oneflow::BlobDesc* (std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)>, oneflow::ParallelContext const*) const () from /data/home/lixiang/oneflow/build/liboneflow.so
#2  0x00007fffd8708cf8 in oneflow::Operator::InferBlobDescsIf(std::function<oneflow::BlobDesc* (std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)> const&, oneflow::ParallelContext const*, oneflow::JobDesc const*) const ()
   from /data/home/lixiang/oneflow/build/liboneflow.so
#3  0x00007fffd7cb536f in oneflow::ExecNode::InferBlobDescs(oneflow::ParallelContext const*) ()
   from /data/home/lixiang/oneflow/build/liboneflow.so
#4  0x00007fffd7d00c1e in oneflow::SliceBoxingTaskNode::BuildExecGphAndRegst() () from /data/home/lixiang/oneflow/build/liboneflow.so
#5  0x00007fffd7d077f7 in std::_Function_handler<oneflow::Maybe<void, void> (oneflow::TaskNode*), oneflow::Graph<oneflow::TaskNode, oneflow::TaskEdge>::TopoForEachNode(std::function<void (oneflow::TaskNode*)>) const::{lambda(char const*)#1}::operator()(char const*) const::{lambda(oneflow::TaskNode*)#1}>::_M_invoke(std::_Any_data const&, oneflow::TaskNode*&&) () from /data/home/lixiang/oneflow/build/liboneflow.so
#6  0x00007fffd7d08478 in oneflow::Graph<oneflow::TaskNode, oneflow::TaskEdge>::TopoForEachNodeWithErrorCaptured(std::function<void (oneflow::TaskNode*, std::function<void (oneflow::TaskNode*)> const&)> const&, std::function<void (oneflow::TaskNode*, std::function<void (oneflow::TaskNode*)> const&)> const&, std::function<oneflow::Maybe<void, void> (oneflow::TaskNode*)> const&) const ()
--Type <RET> for more, q to quit, c to continue without paging--
   from /data/home/lixiang/oneflow/build/liboneflow.so
#7  0x00007fffd7d093e7 in oneflow::Graph<oneflow::TaskNode, oneflow::TaskEdge>::TopoForEachNode(std::function<void (oneflow::TaskNode*)>) const::{lambda(char const*)#1}::operator()(char const*) const () from /data/home/lixiang/oneflow/build/liboneflow.so
#8  0x00007fffd7d938cd in oneflow::Compiler::Compile(oneflow::Job*, oneflow::Plan*) const ()
   from /data/home/lixiang/oneflow/build/liboneflow.so
#9  0x00007fffd7343c4b in oneflow::NNGraph::CompilePlanForRuntime() () from /data/home/lixiang/oneflow/build/liboneflow.so
#10 0x00007ffff7337b8e in ?? () from /data/home/lixiang/oneflow/python/oneflow/_oneflow_internal.cpython-38-x86_64-linux-gnu.so
#11 0x00007ffff7291cc4 in ?? () from /data/home/lixiang/oneflow/python/oneflow/_oneflow_internal.cpython-38-x86_64-linux-gnu.so
#12 0x00005555556d3706 in cfunction_call_varargs (kwargs=<optimized out>, args=<optimized out>, func=0x7fff60030220)
    at /tmp/build/80754af9/python_1593706424329/work/Objects/call.c:742
#13 PyCFunction_Call (func=0x7fff60030220, args=<optimized out>, kwargs=<optimized out>)
    at /tmp/build/80754af9/python_1593706424329/work/Objects/call.c:772
#14 0x000055555569125f in _PyObject_MakeTpCall (callable=0x7fff60030220, args=<optimized out>, nargs=<optimized out>, 
    keywords=<optimized out>) at /tmp/build/80754af9/python_1593706424329/work/Objects/call.c:159
#15 0x00005555556df8a1 in _PyObject_Vectorcall (kwnames=<optimized out>, nargsf=1, args=0x5555816e1ba0, callable=0x7fff60030220)
    at /tmp/build/80754af9/python_1593706424329/work/Include/cpython/abstract.h:125
#16 method_vectorcall (method=<optimized out>, args=0x5555816e1ba8, nargsf=<optimized out>, kwnames=<optimized out>)
--Type <RET> for more, q to quit, c to continue without paging--
    at /tmp/build/80754af9/python_1593706424329/work/Objects/classobject.c:60
#17 0x000055555565358e in _PyObject_Vectorcall (kwnames=0x0, nargsf=<optimized out>, args=0x5555816e1ba8, callable=0x7fff5dc97e40)
    at /tmp/build/80754af9/python_1593706424329/work/Include/cpython/abstract.h:127
#18 call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, tstate=0x5555558ea2e0)
    at /tmp/build/80754af9/python_1593706424329/work/Python/ceval.c:4963
#19 _PyEval_EvalFrameDefault (f=<optimized out>, throwflag=<optimized out>)
    at /tmp/build/80754af9/python_1593706424329/work/Python/ceval.c:3469
#20 0x00007ffff74db943 in oneflow::RecordAndEvalFrame(_frame*, int) ()
   from /data/home/lixiang/oneflow/python/oneflow/_oneflow_internal.cpython-38-x86_64-linux-gnu.so
#21 0x00005555556df50b in function_code_fastcall (globals=<optimized out>, nargs=1, args=<optimized out>, co=<optimized out>)
    at /tmp/build/80754af9/python_1593706424329/work/Objects/call.c:283
#22 _PyFunction_Vectorcall (kwnames=<optimized out>, nargsf=<optimized out>, stack=0x7ffe8ec12748, func=0x7fff5db6a8b0)
    at /tmp/build/80754af9/python_1593706424329/work/Objects/call.c:410
#23 _PyObject_Vectorcall (kwnames=<optimized out>, nargsf=<optimized out>, args=0x7ffe8ec12748, callable=0x7fff5db6a8b0)
    at /tmp/build/80754af9/python_1593706424329/work/Include/cpython/abstract.h:127
#24 method_vectorcall (method=<optimized out>, args=0x7ffe8ec12750, nargsf=<optimized out>, kwnames=<optimized out>)
    at /tmp/build/80754af9/python_1593706424329/work/Objects/classobject.c:60
--Type <RET> for more, q to quit, c to continue without paging--
#25 0x000055555565358e in _PyObject_Vectorcall (kwnames=0x0, nargsf=<optimized out>, args=0x7ffe8ec12750, callable=0x7ffe91055480)
    at /tmp/build/80754af9/python_1593706424329/work/Include/cpython/abstract.h:127
#26 call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, tstate=0x5555558ea2e0)
    at /tmp/build/80754af9/python_1593706424329/work/Python/ceval.c:4963
#27 _PyEval_EvalFrameDefault (f=<optimized out>, throwflag=<optimized out>)
    at /tmp/build/80754af9/python_1593706424329/work/Python/ceval.c:3469
#28 0x00007ffff74db943 in oneflow::RecordAndEvalFrame(_frame*, int) ()
   from /data/home/lixiang/oneflow/python/oneflow/_oneflow_internal.cpython-38-x86_64-linux-gnu.so
#29 0x00005555556de2a2 in _PyEval_EvalCodeWithName (_co=0x7fff5dc67be0, globals=<optimized out>, locals=<optimized out>, 
    args=<optimized out>, argcount=<optimized out>, kwnames=0x0, kwargs=0x7fffffffd4b0, kwcount=<optimized out>, kwstep=1, defs=0x0, 
    defcount=0, kwdefs=0x0, closure=0x0, name=0x7fff5ffe34f0, qualname=0x7fff5dc66cf0)
    at /tmp/build/80754af9/python_1593706424329/work/Python/ceval.c:4298
#30 0x00005555556df243 in _PyFunction_Vectorcall (func=<optimized out>, stack=0x7fffffffd4a0, nargsf=<optimized out>, kwnames=<optimized out>)
    at /tmp/build/80754af9/python_1593706424329/work/Objects/call.c:435
#31 0x00005555556df779 in _PyObject_Vectorcall (kwnames=0x0, nargsf=2, args=0x7fffffffd4a0, callable=0x7fff5db6a430)
    at /tmp/build/80754af9/python_1593706424329/work/Include/cpython/abstract.h:127
#32 method_vectorcall (method=<optimized out>, args=<optimized out>, nargsf=<optimized out>, kwnames=<optimized out>)
--Type <RET> for more, q to quit, c to continue without paging--
    at /tmp/build/80754af9/python_1593706424329/work/Objects/classobject.c:89
#33 0x0000555555690a41 in PyVectorcall_Call (callable=0x7ffff79093c0, tuple=<optimized out>, kwargs=<optimized out>)
    at /tmp/build/80754af9/python_1593706424329/work/Objects/call.c:199
#34 0x00005555557171dc in do_call_core (kwdict=0x7ffe90e6d640, callargs=0x7ffff7883be0, func=0x7ffff79093c0, tstate=<optimized out>)
    at /tmp/build/80754af9/python_1593706424329/work/Python/ceval.c:5010
#35 _PyEval_EvalFrameDefault (f=<optimized out>, throwflag=<optimized out>)
    at /tmp/build/80754af9/python_1593706424329/work/Python/ceval.c:3559
#36 0x00007ffff74db943 in oneflow::RecordAndEvalFrame(_frame*, int) ()
   from /data/home/lixiang/oneflow/python/oneflow/_oneflow_internal.cpython-38-x86_64-linux-gnu.so
#37 0x00005555556de7af in _PyEval_EvalCodeWithName (_co=0x7fff5dcd2df0, globals=<optimized out>, locals=<optimized out>, 
    args=<optimized out>, argcount=<optimized out>, kwnames=0x0, kwargs=0x7fffffffd880, kwcount=<optimized out>, kwstep=1, defs=0x0, 
    defcount=0, kwdefs=0x0, closure=0x0, name=0x7ffff79191f0, qualname=0x7fff5dc636f0)
    at /tmp/build/80754af9/python_1593706424329/work/Python/ceval.c:4298
#38 0x00005555556dfd8c in _PyFunction_Vectorcall (kwnames=0x0, nargsf=<optimized out>, stack=0x7fffffffd870, func=0x7fff5db67430)
    at /tmp/build/80754af9/python_1593706424329/work/Objects/call.c:435
#39 _PyObject_FastCallDict (callable=0x7fff5db67430, args=0x7fffffffd870, nargsf=<optimized out>, kwargs=<optimized out>)
    at /tmp/build/80754af9/python_1593706424329/work/Objects/call.c:96
--Type <RET> for more, q to quit, c to continue without paging--
#40 0x00005555556e0033 in _PyObject_Call_Prepend (callable=0x7fff5db67430, obj=<optimized out>, args=0x7ffff7853790, kwargs=0x0)
    at /tmp/build/80754af9/python_1593706424329/work/Objects/call.c:887
#41 0x00005555556e013a in slot_tp_call (self=0x7ffff7796b80, args=0x7ffff7853790, kwds=0x0)
    at /tmp/build/80754af9/python_1593706424329/work/Objects/typeobject.c:6521
#42 0x000055555569125f in _PyObject_MakeTpCall (callable=0x7ffff7796b80, args=<optimized out>, nargs=<optimized out>, 
    keywords=<optimized out>) at /tmp/build/80754af9/python_1593706424329/work/Objects/call.c:159
#43 0x00005555557155e5 in _PyObject_Vectorcall (kwnames=0x0, nargsf=<optimized out>, args=0x7ffff78655b8, callable=<optimized out>)
    at /tmp/build/80754af9/python_1593706424329/work/Include/cpython/abstract.h:125
#44 call_function (kwnames=0x0, oparg=<optimized out>, pp_stack=<synthetic pointer>, tstate=0x5555558ea2e0)
    at /tmp/build/80754af9/python_1593706424329/work/Python/ceval.c:4963
#45 _PyEval_EvalFrameDefault (f=<optimized out>, throwflag=<optimized out>)
    at /tmp/build/80754af9/python_1593706424329/work/Python/ceval.c:3500
#46 0x00005555556de2a2 in _PyEval_EvalCodeWithName (_co=0x7ffff77773a0, globals=<optimized out>, locals=<optimized out>, 
    args=<optimized out>, argcount=<optimized out>, kwnames=0x0, kwargs=0x0, kwcount=<optimized out>, kwstep=2, defs=0x0, defcount=0, 
    kwdefs=0x0, closure=0x0, name=0x0, qualname=0x0) at /tmp/build/80754af9/python_1593706424329/work/Python/ceval.c:4298
#47 0x00005555556df054 in PyEval_EvalCodeEx (_co=<optimized out>, globals=<optimized out>, locals=<optimized out>, args=<optimized out>, 
    argcount=<optimized out>, kws=<optimized out>, kwcount=0, defs=0x0, defcount=0, kwdefs=0x0, closure=0x0)
--Type <RET> for more, q to quit, c to continue without paging--
    at /tmp/build/80754af9/python_1593706424329/work/Python/ceval.c:4327
#48 0x000055555576d5bc in PyEval_EvalCode (co=<optimized out>, globals=<optimized out>, locals=<optimized out>)
    at /tmp/build/80754af9/python_1593706424329/work/Python/ceval.c:718
#49 0x000055555576d664 in run_eval_code_obj (co=0x7ffff77773a0, globals=0x7ffff78c6d00, locals=0x7ffff78c6d00)
    at /tmp/build/80754af9/python_1593706424329/work/Python/pythonrun.c:1125
#50 0x000055555579f874 in run_mod (mod=<optimized out>, filename=<optimized out>, globals=0x7ffff78c6d00, locals=0x7ffff78c6d00, 
    flags=<optimized out>, arena=<optimized out>) at /tmp/build/80754af9/python_1593706424329/work/Python/pythonrun.c:1147
#51 0x0000555555668316 in PyRun_FileExFlags (fp=0x55555592a250, filename_str=<optimized out>, start=<optimized out>, globals=0x7ffff78c6d00, 
    locals=0x7ffff78c6d00, closeit=1, flags=0x7fffffffdd18) at /tmp/build/80754af9/python_1593706424329/work/Python/pythonrun.c:1063
#52 0x00005555556686fb in PyRun_SimpleFileExFlags (fp=0x55555592a250, filename=<optimized out>, closeit=1, flags=0x7fffffffdd18)
    at /tmp/build/80754af9/python_1593706424329/work/Python/pythonrun.c:428
#53 0x00005555556695c0 in pymain_run_file (cf=0x7fffffffdd18, config=0x5555558e93e0)
    at /tmp/build/80754af9/python_1593706424329/work/Modules/main.c:387
#54 pymain_run_python (exitcode=0x7fffffffdd10) at /tmp/build/80754af9/python_1593706424329/work/Modules/main.c:571
#55 Py_RunMain () at /tmp/build/80754af9/python_1593706424329/work/Modules/main.c:650
#56 0x00005555557a2389 in Py_BytesMain (argc=<optimized out>, argv=<optimized out>)
    at /tmp/build/80754af9/python_1593706424329/work/Modules/main.c:1096
--Type <RET> for more, q to quit, c to continue without paging--
#57 0x00007ffff7dcb083 in __libc_start_main (main=0x55555566a040 <main>, argc=3, argv=0x7fffffffdf18, init=<optimized out>, 
    fini=<optimized out>, rtld_fini=<optimized out>, stack_end=0x7fffffffdf08) at ../csu/libc-start.c:308
#58 0x0000555555732553 in _start () at ../sysdeps/x86_64/elf/start.S:103
  </code></pre>
</details>
